src/Elm/Parser/Tokens.elm

```diff
@@ -166,14 +165,14 @@ stringLiteral : Core.Parser String
 stringLiteral =
     Core.succeed identity
         |. Core.symbol "\""
-        |= Core.loop [] stringLiteralHelper
+        |= Core.loop "" stringLiteralHelper
 
 
-stringLiteralHelper : List String -> Core.Parser (Step (List String) String)
-stringLiteralHelper partsSoFarReverse =
+stringLiteralHelper : String -> Core.Parser (Step String String)
+stringLiteralHelper stringSoFar =
     Core.oneOf
-        [ Core.symbol "\"" |> Core.map (\() -> Done (partsSoFarReverse |> List.Extra.listReverseThenStringConcatFast))
-        , Core.succeed (\v -> Loop (String.fromChar v :: partsSoFarReverse))
+        [ Core.symbol "\"" |> Core.map (\() -> Done stringSoFar)
+        , Core.succeed (\v -> Loop (stringSoFar ++ String.fromChar v))
```

Comment:
In practice, maybe we should do `stringSoFar ++ String.fromChar v ++ ""`, that sometimes improves performance.
(The reasoning: `++` can be used for lists and arrays, and by default Elm doesn't know which one, so it uses some kind of `append` function here. If however there is a very explicit string in the `++` chain, then it knows, and uses JS' `+` operator.)